<?xml version="1.0" encoding="UTF-8"?>
<!-- Launches the offb_node in the correct namespace -->
<launch>
    <!-- drone_count can be inaccurate, will be automatically set anyway -->
    <param name="/drone_count" value="0" />
    <!-- Make sure that drone_number starts from 0 and is continuous -->
    <!-- It is used to count the total number of drones in the swarm -->
    <include file="$(find carriers)/launch/swarm_member.launch">
        <arg name="drone_number" value="0"/>
    </include>

    <include file="$(find carriers)/launch/swarm_member.launch">
        <arg name="drone_number" value="1"/>
    </include>

    <include file="$(find carriers)/launch/swarm_member.launch">
        <arg name="drone_number" value="2"/>
    </include>
    <!-- This node finds the centre of mass of all the drones in the swarm -->
    <node pkg="carriers" type="centre_of_mass_server" name="centre_of_mass_server" output="screen"/>

    <node pkg="carriers" type="main" name="main" output="screen"/>
</launch>
